node/main-ipc.ts

```diff
@@ -0,0 +1,353 @@
+
+    finalArray.forEach((element: ImageElement) => {
+      allHashes.set(element.hash, 1);
+    });
+    removeThumbnailsNotInHub(allHashes, screenshotOutputFolder);
+  });
+
+  /**
+   * Summon system modal to choose OUTPUT directory
+   * where the final .vha2 file, vha-folder, and all screenshots will be saved
+   */
+  ipc.on('choose-output', (event) => {
+    dialog.showOpenDialog(win, {
+      properties: ['openDirectory']
+    }).then(result => {
+      const outputDirPath: string = result.filePaths[0];
+      if (outputDirPath) {
+        event.sender.send('output-folder-chosen', outputDirPath);
+      }
+    }).catch(err => {});
+  });
+
+  /**
+   * Try to rename the particular file
+   */
+  ipc.on('try-to-rename-this-file', (event, sourceFolder: string, relPath: string, file: string, renameTo: string, index: number): void => {
+    console.log('renaming file:');
+
+    const original: string = path.join(sourceFolder, relPath, file);
+    const newName: string = path.join(sourceFolder, relPath, renameTo);
+
+    console.log(original);
+    console.log(newName);
+
+    let success = true;
+    let errMsg: string;
+
+    // check if already exists first
+    if (fs.existsSync(newName)) {
+      console.log('some file already EXISTS WITH THAT NAME !!!');
+      success = false;
+      errMsg = 'RIGHCLICK.errorFileNameExists';
+    } else {
+      try {
+        fs.renameSync(original, newName);
+      } catch (err) {
+        success = false;
+        console.log(err);
+        if (err.code === 'ENOENT') {
+          // const pathObj = path.parse(err.path);
+          // console.log(pathObj);
+          errMsg = 'RIGHTCLICK.errorFileNotFound';
+        } else {
+          errMsg = 'RIGHTCLICK.errorSomeError';
+        }
+      }
+    }
+
+    event.sender.send('rename-file-response', index, success, renameTo, file, errMsg);
+  });
+
+  /**
+   * Close the window / quit / exit the app
+   */
+  ipc.on('close-window', (event, settingsToSave: SettingsObject, finalObjectToSave: FinalObject) => {
+
+    // convert shortcuts map to object
+    // someday when node stops giving error: Property 'fromEntries' does not exist on type 'ObjectConstructor'
+    // settingsToSave.shortcuts = <any>Object.fromEntries(settingsToSave.shortcuts);
+    // until then: https://gist.github.com/lukehorvat/133e2293ba6ae96a35ba#gistcomment-2600839
+    let obj = Array.from(settingsToSave.shortcuts).reduce((obj, [key, value]) => {
+      obj[key] = value;
+      return obj;
+    }, {});
+    settingsToSave.shortcuts = <any>obj;
+
+    const json = JSON.stringify(settingsToSave);
+
+    try {
+      fs.statSync(path.join(pathToAppData, 'video-hub-app-2'));
+    } catch (e) {
+      fs.mkdirSync(path.join(pathToAppData, 'video-hub-app-2'));
+    }
+
+    // TODO -- catch bug if user closes before selecting the output folder ?!??
+    fs.writeFile(path.join(pathToAppData, 'video-hub-app-2', 'settings.json'), json, 'utf8', () => {
```

Comment:
`video-hub-app-3`?

Maybe import and clone vha2 settings if they exist, and save to vha3 settings?